### PR TITLE
getFoodTypes working

### PR DESCRIPTION
**Pull request by PatriceBriggs**
Hi James,

Thanks for your earlier input - that helps.  A few more questions:

I'm working on the getFoodTypes using _.union.  I'm having trouble getting it to work as expected.  I created an array of all the food types, then used that array in the _.union.  Is there a different way that I should pass the information in?  Maybe this isn't what you mean for us to do?  Are you looking for just the list of foods for each truck?

Got the /trucks route figured out.  Still trying to figure out the _.union.

Thanks,
Patrice

### CODE REVIEW

**Comment by jamdon** [comment]
1. In this case, you don't need an Express folder. The only folders you needed were the root project folder, and a public folder inside that. The app.js and trucks.js files can go in the root project folder. The index.html, client.js, and any other static files (e.g. html, css, and client js files) all go in the public directory.
2. No, underscore isn't necessary to return all the trucks - just return them. You'll need underscore for some of the other functions, but getTrucks just returns the foodTrucks array.
3. No, there's no functionality in this assignment requiring enhancedDate.
4. Yes, next is used in middleware. The only middleware you're using in this assignment, however, is the express.static middleware, and it calls next internally, so you don't need to worry about it.

**Comment by jamdon** [comment]
To get the list of unique food types, there's a two-step process:
1. compile the list of all food types for all trucks - you can either loop through all the trucks and for each truck, loop through the food types, adding them to a list. Each truck you loop over can just keep adding to this list. Another way to approach this is to map over the food truck list, returning the actual food type array for each truck. Since the map function returns a new array, you'll want to assign the return value to a variable, which will be an array of arrays when you're done. If you go that route, you can "flatten" that resulting array to get just a simple array (see underscore for map and flatten.)
2. create a unique list based on this full list you've compiled. Check out underscore to see how the uniq function works.

So to recap - you're going to want to look at map, flatten, and uniq.

**Comment by jamdon** [comment]
You missed the first step in the assignment: create a package.json file by using `npm init` This step is critical because it allows me (or anyone else) to clone your repository and install any dependencies it requires so I can run your server (and verify it works.) I'm only assuming you installed express locally when you were working on this? That's one of the dependencies you'll want to get into your package.json, and unless you create it first, installing express using `npm install --save express` won't put anything in package.json because it won't exist. Same thing with underscore.

**Comment by jamdon** [comment]
Except for the lack of a package.json file, it works great. :+1:

**Comment by PatriceBriggs** [comment]
Hi James, 
I did the npm init and installed express and there is a package.json file on my local repository.  So this sounds a little silly to ask, but do I have to add the package.json file and commit it as well as the other files?  I didn't do that.  How about the node_modules and the .gitignore.  Do I have to do an add and commit on those before the push?

Thanks!
Patrice
